participant/src/comms/http.rs

```diff
@@ -0,0 +1,132 @@
+//! HTTP implementation of the Comms trait.
+
+use async_trait::async_trait;
+#[cfg(not(feature = "redpallas"))]
+use frost_ed25519 as frost;
+#[cfg(feature = "redpallas")]
+use reddsa::frost::redpallas as frost;
+
+use eyre::eyre;
+
+use frost::{round1::SigningCommitments, round2::SignatureShare, Identifier};
+use reqwest::Response;
+
+use super::{Comms, GenericSigningPackage};
+
+use std::io::{BufRead, Write};
+
+use std::error::Error;
+
+use std::time::Duration;
+
+use crate::args::Args;
+
+pub struct HTTPComms {
+    client: reqwest::Client,
+    host_port: String,
+    session_id: Uuid,
+}
+
+use server::Uuid;
+
+// TODO: Improve error handling for invalid session id
+impl HTTPComms {
+    pub fn new(args: &Args) -> Self {
+        let client = reqwest::Client::new();
+        Self {
+            client,
+            host_port: format!("http://{}:{}", args.ip, args.port),
+            session_id: Uuid::parse_str(&args.session_id).expect("invalid session id"),
+        }
+    }
+}
+
+#[async_trait(?Send)]
+impl Comms for HTTPComms {
+    async fn get_signing_package(
+        &mut self,
+        _input: &mut dyn BufRead,
+        _output: &mut dyn Write,
+        commitments: SigningCommitments,
+        identifier: Identifier,
+        session_id: Uuid,
+    ) -> Result<GenericSigningPackage, Box<dyn Error>> {
+        // Send Commitments to Server
+        let r = self
+            .client
+            .post(format!("{}/send_commitments", self.host_port))
+            .json(&server::SendCommitmentsArgs {
+                session_id,
+                identifier,
+                commitments: vec![commitments],
+            })
+            .send()
+            .await?;
+
+        println!("commitments: {:?}", Response::text(r).await.unwrap());
+
+        eprint!("Waiting for coordinator to send signing package...");
+
+        // Receive SigningPackage from Coordinator
+
+        let r = loop {
+            let r = self
+                .client
+                .post(format!("{}/get_signing_package", self.host_port))
+                .json(&server::GetSigningPackageArgs { session_id })
+                .send()
+                .await?;
+            if r.status() != 200 {
+                tokio::time::sleep(Duration::from_secs(2)).await;
+                eprint!(".");
+            } else {
+                break r.json::<server::GetSigningPackageOutput>().await?;
+            }
+            eprintln!("Signing package received");
```

Comment:
This is currently printing "Signing package received" once per second.

```suggestion
            } else {
                eprintln!("\nSigning package received");
                break r.json::<server::GetSigningPackageOutput>().await?;
            }
```

Reply:
Haha, yes not ideal. Will fix